moped-database/migrations/1704919187107_moped_proj_funding_rename_columns/up.sql

```diff
@@ -0,0 +1,34 @@
+ALTER TABLE moped_proj_funding RENAME COLUMN date_added TO created_at;
+ALTER TABLE moped_proj_funding RENAME COLUMN added_by TO created_by_user_id;
+
+ALTER TABLE moped_proj_funding ADD COLUMN updated_at TIMESTAMP WITH TIME ZONE;
+ALTER TABLE moped_proj_funding ADD COLUMN updated_by_user_id INTEGER;
+
+COMMENT ON COLUMN moped_proj_funding.updated_at IS 'Timestamp when the record was last updated';
+COMMENT ON COLUMN moped_proj_funding.updated_by_user_id IS 'ID of the user who last updated the record';
+
+CREATE OR REPLACE FUNCTION public.update_self_and_project_updated_audit_fields() RETURNS TRIGGER
+    LANGUAGE plpgsql
+    AS $$
+BEGIN
+    -- Update the updated_at field in the current row of the triggering table
+    NEW.updated_at := NOW();
+
+    -- Update the updated_at and updated_by_user_id fields in the related row of moped_project
+    UPDATE moped_project
+    SET updated_at = NOW(),
+        updated_by_user_id = NEW.updated_by_user_id
+    WHERE project_id = NEW.project_id;
+
+    RETURN NEW;
+END;
+$$;
```

Comment:
In the issue for this work, [16086](https://github.com/cityofaustin/atd-data-tech/issues/15086), wisely points out the consideration that we already have a couple of functions in our database related to these audit update fields, and it will be messy and hard to maintain if we end up making a whole bunch of them, especially if the names don't follow a scheme etc. I recognize that consideration and here's what I'm thinking in this PR and going forward. 

* `update_parent_records_audit_logs()` exists, and when it's triggered to run, it wants to update a parent `moped_proj_component()` record and grandparent `moped_project` record.
* `set_updated_at()` exists, and when it runs, it wants to update _self_, meaning whatever table triggered the f unction call. 
* This PR creates `update_self_and_project_updated_audit_fields()` which updates _self_ and a parent `moped_project` record.

I think if you think about these functions in the context of what chain of tables they are going to take to get back up to moped_project, you will have discovered all the cases for which we're going to create a new function. Neither of the existing functions covered the "self + `moped_project`" case, so a new function was needed, but this new function should work for all the tables that are one step away from `moped_project`.

Reply:
> This PR creates update_self_and_project_updated_audit_fields() which updates self and a parent moped_project record.

Got it—perfect! 

> `set_updated_at()` exists

We will be able to remove that function by re-configuring the `moped_proj_work_activities` table to match what you've done here. I will make sure that gets captured in a follow-up issue 🙏 


> we're going to need a new function for every combination of tables that are required to get back up to `moped_projects`

I have an idea for handling these cases that I shared in Slack. Since that "bubbling up" is not a concern for this PR, I will make sure to capture a potential follow-up in a separate issue.